frontend/server/handlers/artifacts.ts

```diff
@@ -74,6 +76,79 @@ export interface GCSProviderInfo {
   };
 }
 
+/**
+ * Returns an authorization middleware for artifact endpoints.
+ * This middleware handles 3 modes:
+ *
+ * 1. Standalone KFP deployment without Kubeflow platform (single-tenant): No namespace parameter, no Subject access review, 100 % insecure
+ * 2. Default multi-tenant deployment of KFP within Kubeflow platform: No namespace parameter, just verify that the user is authenticated to access the artifact from the specific namespace folder on the object storage via Subject access review.
+ * 3. Artifact PROXY MODE (multi-tenant): Namespace parameter provided - validate format and check RBAC, adds significant overhead to each namespace, decreases scalability and is prone to many CVEs in the artifact proxy deployment
```

Comment:
See the comment above https://github.com/kubeflow/pipelines/pull/12550#issuecomment-3693336964